GitUI/CommandsDialogs/RepoHosting/ViewPullRequestsForm.cs

```diff
@@ -73,22 +73,27 @@ private void ViewPullRequestsForm_Load(object sender, EventArgs e)
             _loader.LoadAsync(
                 () =>
                 {
-                    var t = _gitHoster.GetHostedRemotesForModule().ToList();
-                    foreach (var el in t)
+                    var hostedRemotes = _gitHoster.GetHostedRemotesForModule().ToArray();
+
+                    // load all hosted repositories.
+                    foreach (var hostedRemote in hostedRemotes)
                     {
-                        el.GetHostedRepository(); // We do this now because we want to do it in the async part.
+                        try
+                        {
+                            hostedRemote.GetHostedRepository(); // We do this now because we want to do it in the async part.
+                        }
+                        catch (Exception)
```

Comment:
```
catch
{}
```

Reply:
Why?

Reply:
In theory in IL it's possible to throw an arbitrary object. Catching `Exception` and an unspecified catch are technically different. I have never seen it make a difference in practice.

This is another example of something where an analyzer should step in to enfore this kind of thing according to the project's preferences IMO.